File: backend/src/main/kotlin/com/josoder/backend/model/CountryStats.kt
package com.josoder.backend.model

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document


@Document(collection = "countrystats")
class CountryStats(
        @Id
        val country: String,
        val cases: Long,
        val todayCases: Long,
        val deaths: Long,
        val todayDeaths: Long,
        val recovered: Long,
        val active: Long,
        val critical: Long,
        val casesPerOneMillion: Long,
        val deathsPerOneMillion: Long,
        val updated: Long)

data class CountryStatsDTO(val country: String,
                           val cases: Long,
                           val todayCases: Long,
                           val deaths: Long,
                           val todayDeaths: Long,
                           val recovered: Long,
                           val active: Long,
                           val critical: Long,
                           val casesPerOneMillion: Long,
                           val deathsPerOneMillion: Long,
                           val updated: Long) {
    fun convertToEntity(): CountryStats {
        return CountryStats(this.country, this.cases, this.todayCases, this.deaths, this.todayDeaths, this.recovered,
                this.active, this.critical, this.casesPerOneMillion, this.deathsPerOneMillion, this.updated)
    }
}

File: frontend/src/app/feature/stats/model/country-stats.ts
export class CountryStats {
  public static readonly GLOBAL = 'global';
  country: string;
  cases: number;
  todayCases: number;
  deaths: number;
  todayDeaths: number;
  recovered: number;
  active: number;
  critical: number;
  casesPerOneMillion: number;
  deathsPerOneMillion: number;
  updated: number;


  constructor(country?: string, cases?: number, todayCases?: number, deaths?: number, todayDeaths?: number,
              recovered?: number, active?: number, critical?: number, casesPerOneMillion?: number,
              deathsPerOneMillion?: number, updated?: number) {
    this.country = country;
    this.cases = cases;
    this.todayCases = todayCases;
    this.deaths = deaths;
    this.todayDeaths = todayDeaths;
    this.recovered = recovered;
    this.active = active;
    this.critical = critical;
    this.casesPerOneMillion = casesPerOneMillion;
    this.deathsPerOneMillion = deathsPerOneMillion;
    this.updated = updated;
  }
}
